asyn/asynPortDriver/asynPortDriver.cpp

```diff
@@ -3921,10 +3947,30 @@ asynStatus asynPortDriver::createParams()
     return asynSuccess;
 }
 
+asynStatus asynPortDriver::shutdown() {
+    shutdownNeeded = false;
+    delete cbThread;
+    cbThread = NULL;
+    return asynSuccess;
+}
+
 /** Destructor for asynPortDriver class; frees resources allocated when port driver is created. */
 asynPortDriver::~asynPortDriver()
 {
-    delete cbThread;
+    if (shutdownNeeded) {
+        // If shutdownNeeded is still true, it means that the base
+        // implementation of asynPortDriver::shutdown() was not called. Which
+        // means that one of the derived classes does not call the base
+        // function, and needs fixing.
+        asynPrint(this->pasynUserSelf, ASYN_TRACE_ERROR,
+                  "%s: port=%s Driver error: port supports shutdown, but it was "
+                  "incomplete. A derived class of asynPortDriver has a bug.\n",
+                  driverName, portName);
+    }
+
+    if (cbThread)
+        delete cbThread;
```

Comment:
AFAIK it's safe to delete a nullptr, so you shouldn't need the conditional.

I see now this was removed in a following commit.